cmake/ProcessOptions.cmake

```diff
@@ -22,13 +22,14 @@
 # add custom warnings and optimizations
 function( NEST_PROCESS_WITH_OPTIMIZE )
   if ( with-optimize )
-    if ( with-optimize STREQUAL "ON" )
+  string(TOUPPER "${with-optimize}" WITHOPTIMIZE)
+    if ( WITHOPTIMIZE STREQUAL "ON" )
       set( with-optimize "-O2" )
     endif ()
-    foreach ( flag ${with-optimize} )
-      set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${flag}" PARENT_SCOPE )
-      set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flag}" PARENT_SCOPE )
-    endforeach ()
+    set(OPTIMIZATION_FLAGS "")
+    string(JOIN " " OPTIMIZATION_FLAGS  ${with-optimize} )
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OPTIMIZATION_FLAGS}" PARENT_SCOPE )
+    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OPTIMIZATION_FLAGS}" PARENT_SCOPE )
```

Comment:
That's a much cleaner solution! Thanks for optimizing the optimization flags :grin: